src/openrct2-ui/interface/ViewportInteraction.cpp

```diff
@@ -80,7 +80,7 @@ int32_t viewport_interaction_get_item_left(int32_t x, int32_t y, viewport_intera
         if (info->type == VIEWPORT_INTERACTION_ITEM_SPRITE && sprite->generic.sprite_identifier == SPRITE_IDENTIFIER_MISC)
             return info->type;
         else
-            return info->type = VIEWPORT_INTERACTION_ITEM_NONE;
+            return VIEWPORT_INTERACTION_ITEM_NONE;
```

Comment:
Are you sure this assignment is unnecessary? It couild be that the caller doesn't check the return type of this function, but does check the `info` instance.

Reply:
I should have documented how the code got to this point. I don't clearly remember why, but I believe I determined that the assignment was not only unnecessary, but _could_ actually cause a problem.
I'll look into it again and document it this time. It'll be a good exercise for me. Might even reach a different conclusion.

Reply:
@XplosiveLugnut Well, was it? ;-)

Reply:
Heh, I suppose.

OK, so, the only thing the assignment can affect is the _window_title_command_editor_tool_down_ function.
This function doesn't use the return value, only the assigned value.

However, the Title Command Editor isn't usable in the title screen, so this doesn't matter.

In the other two cases where _viewport_interaction_get_item_left_ is used, returning _VIEWPORT_INTERACTION_ITEM_NONE_ causes the calling function to do nothing but return and never use the assigned value. So the lack of assignment doesn't matter here either.

TLDR: There _should_ be no noticeable difference between having or not having the assignment, since the only function that it could affect can't be called while in the title screen.

Reply:
I'd prefer a stricter check for balloons or ducks specifically. (I think we have functions like IsBalloon and IsDuck for that). While there aren't any other misc sprites one can interact with _now_, there is no guarantee there won't be in the future, and they might not be suitable for use in the title screen.

Reply:
Still, doing it like this (which is more in line with how the code was) shouldn't cause problems, right?

```suggestion
            return info->type = VIEWPORT_INTERACTION_ITEM_NONE;
```